IBMPowerVSCluster runtime error: invalid memory address or nil pointer dereference for OSA region

/kind bug
/area provider/ibmcloud

**What steps did you take and what happened:**
capi-provider pod is restarting with segmentation fault. 

```
oc get pods -n clusters-example       
NAME                                             READY   STATUS             RESTARTS          AGE
capi-provider-7679cc9549-vs2fd                   1/1     Running            233 (5m10s ago)   20h
catalog-operator-56554594dc-rhl9t                2/2     Running            0                 20h
certified-operators-catalog-57c844c7bf-cjf2v     1/1     Running            0                 20h
cluster-api-85d84b8b84-6rw69                     1/1     Running            0                 20h
cluster-autoscaler-dc9f889c4-96stm               1/1     Running            0                 20h
cluster-policy-controller-55c9fd85bb-mfv7m       1/1     Running            0                 20h
cluster-version-operator-6bd989855b-8drhv        1/1     Running            0                 20h
community-operators-catalog-6c5956749-p9g6m      1/1     Running            0                 20h
control-plane-operator-6cdbbf79b6-445l4          1/1     Running            1 (11h ago)       20h
etcd-0                                           1/1     Running            0                 20h
```

capi-provider logs
```
oc logs capi-provider-7679cc9549-vs2fd -n clusters-example
I0408 03:59:04.029473       1 logr.go:249] setup "msg"="Using v1 version of ProviderID format"
I0408 03:59:04.029785       1 logr.go:249] setup "msg"="Watching cluster-api objects only in namespace for reconciliation"  "namespace"="clusters-example"
I0408 03:59:04.130802       1 request.go:597] Waited for 66.298138ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/flowcontrol.apiserver.k8s.io/v1beta1?timeout=32s
I0408 03:59:04.180802       1 request.go:597] Waited for 116.230391ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/apps.openshift.io/v1?timeout=32s
I0408 03:59:04.230816       1 request.go:597] Waited for 166.256464ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/authorization.openshift.io/v1?timeout=32s
I0408 03:59:04.281338       1 request.go:597] Waited for 216.728698ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/build.openshift.io/v1?timeout=32s
I0408 03:59:04.331253       1 request.go:597] Waited for 266.603398ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/image.openshift.io/v1?timeout=32s
I0408 03:59:04.381146       1 request.go:597] Waited for 316.516615ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/oauth.openshift.io/v1?timeout=32s
I0408 03:59:04.435175       1 request.go:597] Waited for 370.529632ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/project.openshift.io/v1?timeout=32s
I0408 03:59:04.481723       1 request.go:597] Waited for 417.009143ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/quota.openshift.io/v1?timeout=32s
I0408 03:59:04.531327       1 request.go:597] Waited for 466.595628ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/route.openshift.io/v1?timeout=32s
I0408 03:59:04.582251       1 request.go:597] Waited for 517.490204ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/security.openshift.io/v1?timeout=32s
I0408 03:59:04.637303       1 request.go:597] Waited for 572.499489ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/template.openshift.io/v1?timeout=32s
I0408 03:59:04.682641       1 request.go:597] Waited for 617.81513ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/user.openshift.io/v1?timeout=32s
I0408 03:59:04.731125       1 request.go:597] Waited for 666.277504ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/packages.operators.coreos.com/v1?timeout=32s
I0408 03:59:04.781089       1 request.go:597] Waited for 716.241288ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/config.openshift.io/v1?timeout=32s
I0408 03:59:04.834949       1 request.go:597] Waited for 770.082777ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operator.openshift.io/v1?timeout=32s
I0408 03:59:04.881728       1 request.go:597] Waited for 816.80594ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/machineconfiguration.openshift.io/v1?timeout=32s
I0408 03:59:04.930801       1 request.go:597] Waited for 865.877261ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/apiserver.openshift.io/v1?timeout=32s
I0408 03:59:04.986014       1 request.go:597] Waited for 921.046224ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/cloudcredential.openshift.io/v1?timeout=32s
I0408 03:59:05.032114       1 request.go:597] Waited for 967.125073ms due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/console.openshift.io/v1?timeout=32s
I0408 03:59:05.081133       1 request.go:597] Waited for 1.016153495s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/console.openshift.io/v1alpha1?timeout=32s
I0408 03:59:05.081689       1 request.go:665] Waited for 1.016153495s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/console.openshift.io/v1alpha1?timeout=32s
I0408 03:59:05.130760       1 request.go:597] Waited for 1.065763409s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/crd.projectcalico.org/v1?timeout=32s
I0408 03:59:05.181657       1 request.go:597] Waited for 1.1166365s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/imageregistry.operator.openshift.io/v1?timeout=32s
I0408 03:59:05.231726       1 request.go:597] Waited for 1.166704488s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/ingress.operator.openshift.io/v1?timeout=32s
I0408 03:59:05.281781       1 request.go:597] Waited for 1.216685539s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/k8s.cni.cncf.io/v1?timeout=32s
I0408 03:59:05.331063       1 request.go:597] Waited for 1.265940967s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operators.coreos.com/v2?timeout=32s
I0408 03:59:05.381553       1 request.go:597] Waited for 1.316395882s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/monitoring.coreos.com/v1?timeout=32s
I0408 03:59:05.432228       1 request.go:597] Waited for 1.367045008s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/monitoring.coreos.com/v1alpha1?timeout=32s
I0408 03:59:05.480788       1 request.go:597] Waited for 1.415578582s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/network.operator.openshift.io/v1?timeout=32s
I0408 03:59:05.531704       1 request.go:597] Waited for 1.466472188s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operator.tigera.io/v1?timeout=32s
I0408 03:59:05.581197       1 request.go:597] Waited for 1.515948287s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operators.coreos.com/v1alpha2?timeout=32s
I0408 03:59:05.630935       1 request.go:597] Waited for 1.565668335s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operators.coreos.com/v1?timeout=32s
I0408 03:59:05.680986       1 request.go:597] Waited for 1.615697696s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/snapshot.storage.k8s.io/v1?timeout=32s
I0408 03:59:05.731764       1 request.go:597] Waited for 1.666450473s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/snapshot.storage.k8s.io/v1beta1?timeout=32s
I0408 03:59:05.780897       1 request.go:597] Waited for 1.715562171s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/tuned.openshift.io/v1?timeout=32s
I0408 03:59:05.830819       1 request.go:597] Waited for 1.765457603s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/capi-provider.agent-install.openshift.io/v1alpha1?timeout=32s
I0408 03:59:05.881561       1 request.go:597] Waited for 1.816184397s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/controlplane.operator.openshift.io/v1alpha1?timeout=32s
I0408 03:59:05.931416       1 request.go:597] Waited for 1.866025466s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/hypershift.openshift.io/v1alpha1?timeout=32s
I0408 03:59:05.980801       1 request.go:597] Waited for 1.915395792s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/ibm.com/v1alpha1?timeout=32s
I0408 03:59:06.030943       1 request.go:597] Waited for 1.965502588s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/infrastructure.cluster.x-k8s.io/v1beta1?timeout=32s
I0408 03:59:06.083936       1 request.go:597] Waited for 2.018468052s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/infrastructure.cluster.x-k8s.io/v1alpha1?timeout=32s
I0408 03:59:06.084097       1 request.go:665] Waited for 2.018468052s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/infrastructure.cluster.x-k8s.io/v1alpha1?timeout=32s
I0408 03:59:06.131009       1 request.go:597] Waited for 2.065502052s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/migration.k8s.io/v1alpha1?timeout=32s
I0408 03:59:06.181394       1 request.go:597] Waited for 2.115893396s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/whereabouts.cni.cncf.io/v1alpha1?timeout=32s
I0408 03:59:06.231912       1 request.go:597] Waited for 2.166394524s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/addons.cluster.x-k8s.io/v1beta1?timeout=32s
I0408 03:59:06.280938       1 request.go:597] Waited for 2.215397768s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/cluster.x-k8s.io/v1beta1?timeout=32s
I0408 03:59:06.331468       1 request.go:597] Waited for 2.265901796s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/helm.openshift.io/v1beta1?timeout=32s
I0408 03:59:06.381194       1 request.go:597] Waited for 2.315585271s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/operators.coreos.com/v1alpha1?timeout=32s
I0408 03:59:06.431250       1 request.go:597] Waited for 2.36562122s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/samples.operator.openshift.io/v1?timeout=32s
I0408 03:59:06.481374       1 request.go:597] Waited for 2.415624395s due to client-side throttling, not priority and fairness, request: GET:https://172.21.0.1:443/apis/security.internal.openshift.io/v1?timeout=32s
I0408 03:59:06.489574       1 logr.go:249] controller-runtime/metrics "msg"="Metrics server is starting to listen"  "addr"=":8080"
I0408 03:59:06.490912       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSCluster"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervscluster"
I0408 03:59:06.492854       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervscluster"
I0408 03:59:06.492993       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSCluster"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervscluster"
I0408 03:59:06.493116       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervscluster"
I0408 03:59:06.493360       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/convert"
I0408 03:59:06.493446       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSCluster"}
I0408 03:59:06.493487       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachine"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachine"
I0408 03:59:06.493559       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachine"
I0408 03:59:06.493685       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachine"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachine"
I0408 03:59:06.493772       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachine"
I0408 03:59:06.493933       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachine"}
I0408 03:59:06.494034       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachineTemplate"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachinetemplate"
I0408 03:59:06.494331       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachinetemplate"
I0408 03:59:06.494474       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachineTemplate"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachinetemplate"
I0408 03:59:06.494789       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsmachinetemplate"
I0408 03:59:06.495153       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSMachineTemplate"}
I0408 03:59:06.495409       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachine"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachine"
I0408 03:59:06.500217       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachine"
I0408 03:59:06.500354       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachine"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachine"
I0408 03:59:06.500468       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachine"
I0408 03:59:06.500628       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachine"}
I0408 03:59:06.500709       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachineTemplate"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachinetemplate"
I0408 03:59:06.500816       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachinetemplate"
I0408 03:59:06.500940       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachineTemplate"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachinetemplate"
I0408 03:59:06.501064       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpcmachinetemplate"
I0408 03:59:06.501211       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCMachineTemplate"}
I0408 03:59:06.501285       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCCluster"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpccluster"
I0408 03:59:06.501437       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpccluster"
I0408 03:59:06.501552       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCCluster"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpccluster"
I0408 03:59:06.501666       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmvpccluster"
I0408 03:59:06.501818       1 logr.go:249] controller-runtime/builder "msg"="Conversion webhook enabled"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMVPCCluster"}
I0408 03:59:06.501885       1 logr.go:249] controller-runtime/builder "msg"="Registering a mutating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSImage"} "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsimage"
I0408 03:59:06.502015       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/mutate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsimage"
I0408 03:59:06.502148       1 logr.go:249] controller-runtime/builder "msg"="Registering a validating webhook"  "GVK"={"Group":"infrastructure.cluster.x-k8s.io","Version":"v1beta1","Kind":"IBMPowerVSImage"} "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsimage"
I0408 03:59:06.502269       1 server.go:146] controller-runtime/webhook "msg"="Registering webhook" "path"="/validate-infrastructure-cluster-x-k8s-io-v1beta1-ibmpowervsimage"
I0408 03:59:06.502409       1 logr.go:249] setup "msg"="starting manager"
I0408 03:59:06.502708       1 server.go:214] controller-runtime/webhook/webhooks "msg"="Starting webhook server"
I0408 03:59:06.502903       1 internal.go:362]  "msg"="Starting server" "addr"={"IP":"::","Port":8080,"Zone":""} "kind"="metrics" "path"="/metrics"
I0408 03:59:06.503064       1 internal.go:362]  "msg"="Starting server" "addr"={"IP":"::","Port":9440,"Zone":""} "kind"="health probe"
I0408 03:59:06.503239       1 shared_informer.go:270] caches populated
I0408 03:59:06.503480       1 logr.go:249] controller-runtime/certwatcher "msg"="Updated current TLS certificate"
I0408 03:59:06.503513       1 leaderelection.go:248] attempting to acquire leader lease clusters-example/effcf9b8.cluster.x-k8s.io...
I0408 03:59:06.503610       1 logr.go:249] controller-runtime/webhook "msg"="Serving webhook server"  "host"="" "port"=9443
I0408 03:59:06.504153       1 logr.go:249] controller-runtime/certwatcher "msg"="Starting certificate watcher"
I0408 03:59:06.521963       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:06.521987       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:09.424425       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:09.424452       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:13.317621       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:13.317644       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:15.398857       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:15.398881       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:18.219935       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:18.219961       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:21.161607       1 leaderelection.go:352] lock is held by capi-provider-7679cc9549-vs2fd_6e2237d6-0a98-42f3-a6cc-135590e11c5e and has not yet expired
I0408 03:59:21.161642       1 leaderelection.go:253] failed to acquire lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:24.346314       1 leaderelection.go:258] successfully acquired lease clusters-example/effcf9b8.cluster.x-k8s.io
I0408 03:59:24.346593       1 recorder.go:103] events "msg"="Normal"  "message"="capi-provider-7679cc9549-vs2fd_3ff9f335-527c-4885-8a28-26c1589bf41f became leader" "object"={"kind":"ConfigMap","namespace":"clusters-example","name":"effcf9b8.cluster.x-k8s.io","uid":"9540792f-772d-48f7-a9c7-430561f03877","apiVersion":"v1","resourceVersion":"967012"} "reason"="LeaderElection"
I0408 03:59:24.346637       1 recorder.go:103] events "msg"="Normal"  "message"="capi-provider-7679cc9549-vs2fd_3ff9f335-527c-4885-8a28-26c1589bf41f became leader" "object"={"kind":"Lease","namespace":"clusters-example","name":"effcf9b8.cluster.x-k8s.io","uid":"417db2d2-ca0d-42ee-949e-b65cd3bee7ee","apiVersion":"coordination.k8s.io/v1","resourceVersion":"967013"} "reason"="LeaderElection"
I0408 03:59:24.346813       1 controller.go:178] controller/ibmvpccluster "msg"="Starting EventSource" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCCluster" "source"="kind source: *v1beta1.IBMVPCCluster"
I0408 03:59:24.346844       1 controller.go:186] controller/ibmvpccluster "msg"="Starting Controller" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCCluster"
I0408 03:59:24.347289       1 controller.go:178] controller/ibmpowervscluster "msg"="Starting EventSource" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSCluster" "source"="kind source: *v1beta1.IBMPowerVSCluster"
I0408 03:59:24.347376       1 controller.go:186] controller/ibmpowervscluster "msg"="Starting Controller" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSCluster"
I0408 03:59:24.347424       1 controller.go:178] controller/ibmpowervsimage "msg"="Starting EventSource" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSImage" "source"="kind source: *v1beta1.IBMPowerVSImage"
I0408 03:59:24.348092       1 controller.go:186] controller/ibmpowervsimage "msg"="Starting Controller" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSImage"
I0408 03:59:24.347327       1 controller.go:178] controller/ibmvpcmachine "msg"="Starting EventSource" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCMachine" "source"="kind source: *v1beta1.IBMVPCMachine"
I0408 03:59:24.348196       1 controller.go:186] controller/ibmvpcmachine "msg"="Starting Controller" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCMachine"
I0408 03:59:24.347544       1 controller.go:178] controller/ibmpowervsmachine "msg"="Starting EventSource" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSMachine" "source"="kind source: *v1beta1.IBMPowerVSMachine"
I0408 03:59:24.348262       1 controller.go:186] controller/ibmpowervsmachine "msg"="Starting Controller" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSMachine"
I0408 03:59:24.348611       1 reflector.go:219] Starting reflector *v1beta1.IBMVPCCluster (9m41.872290673s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.348630       1 reflector.go:255] Listing and watching *v1beta1.IBMVPCCluster from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.348831       1 reflector.go:219] Starting reflector *v1beta1.IBMVPCMachine (9m39.187156543s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.348850       1 reflector.go:255] Listing and watching *v1beta1.IBMVPCMachine from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.349348       1 reflector.go:219] Starting reflector *v1beta1.IBMPowerVSCluster (10m59.865726948s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.349370       1 reflector.go:255] Listing and watching *v1beta1.IBMPowerVSCluster from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.349879       1 reflector.go:219] Starting reflector *v1beta1.IBMPowerVSImage (9m41.912742295s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.349945       1 reflector.go:255] Listing and watching *v1beta1.IBMPowerVSImage from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.350509       1 reflector.go:219] Starting reflector *v1beta1.IBMPowerVSMachine (9m43.031152663s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.350575       1 reflector.go:255] Listing and watching *v1beta1.IBMPowerVSMachine from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.449287       1 shared_informer.go:270] caches populated
I0408 03:59:24.449324       1 shared_informer.go:270] caches populated
I0408 03:59:24.449359       1 shared_informer.go:270] caches populated
I0408 03:59:24.449361       1 shared_informer.go:270] caches populated
I0408 03:59:24.449396       1 controller.go:220] controller/ibmpowervscluster "msg"="Starting workers" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSCluster" "worker count"=1
I0408 03:59:24.449396       1 controller.go:220] controller/ibmvpcmachine "msg"="Starting workers" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCMachine" "worker count"=1
I0408 03:59:24.449430       1 shared_informer.go:270] caches populated
I0408 03:59:24.449469       1 shared_informer.go:270] caches populated
I0408 03:59:24.449492       1 controller.go:220] controller/ibmvpccluster "msg"="Starting workers" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMVPCCluster" "worker count"=1
I0408 03:59:24.449770       1 reflector.go:219] Starting reflector *v1beta1.Cluster (10m3.471587474s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.449791       1 reflector.go:255] Listing and watching *v1beta1.Cluster from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.450015       1 shared_informer.go:270] caches populated
I0408 03:59:24.450609       1 shared_informer.go:270] caches populated
I0408 03:59:24.549941       1 shared_informer.go:270] caches populated
I0408 03:59:24.552145       1 shared_informer.go:270] caches populated
I0408 03:59:24.552210       1 controller.go:220] controller/ibmpowervsimage "msg"="Starting workers" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSImage" "worker count"=1
I0408 03:59:24.552369       1 shared_informer.go:270] caches populated
I0408 03:59:24.552498       1 controller.go:220] controller/ibmpowervsmachine "msg"="Starting workers" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSMachine" "worker count"=1
I0408 03:59:24.553507       1 reflector.go:219] Starting reflector *v1beta1.Machine (9m18.618024885s) from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.553591       1 reflector.go:255] Listing and watching *v1beta1.Machine from pkg/mod/k8s.io/client-go@v0.23.0/tools/cache/reflector.go:167
I0408 03:59:24.653745       1 shared_informer.go:270] caches populated
E0408 03:59:25.612886       1 controller.go:317] controller/ibmpowervscluster "msg"="Reconciler error" "error"="failed to create scope: region not found for the zone, talk to the developer to add the support into the tool: osa21\nfailed to get region\nsigs.k8s.io/cluster-api-provider-ibmcloud/cloud/scope.NewPowerVSClusterScope\n\t/workspace/cloud/scope/powervs_cluster.go:130\nsigs.k8s.io/cluster-api-provider-ibmcloud/controllers.(*IBMPowerVSClusterReconciler).Reconcile\n\t/workspace/controllers/ibmpowervscluster_controller.go:77\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581" "name"="example" "namespace"="clusters-example" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="IBMPowerVSCluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1704d42]

goroutine 528 [running]:
sigs.k8s.io/cluster-api-provider-ibmcloud/cloud/scope.(*PowerVSImageScope).PatchObject(...)
	/workspace/cloud/scope/powervs_image.go:194
sigs.k8s.io/cluster-api-provider-ibmcloud/cloud/scope.(*PowerVSImageScope).Close(...)
	/workspace/cloud/scope/powervs_image.go:199
sigs.k8s.io/cluster-api-provider-ibmcloud/controllers.(*IBMPowerVSImageReconciler).Reconcile.func1()
	/workspace/controllers/ibmpowervsimage_controller.go:81 +0x22
sigs.k8s.io/cluster-api-provider-ibmcloud/controllers.(*IBMPowerVSImageReconciler).Reconcile(0xc00013d7a0, {0x1ddac38, 0xc000b0e9c0}, {{{0xc0006bb1f0, 0x10}, {0xc00049cf78, 0x15}}})
	/workspace/controllers/ibmpowervsimage_controller.go:92 +0x4a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc00012a000, {0x1ddac38, 0xc000b0e990}, {{{0xc0006bb1f0, 0x1a62cc0}, {0xc00049cf78, 0x413a94}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00012a000, {0x1ddab90, 0xc000786500}, {0x1971960, 0xc000345700})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00012a000, {0x1ddab90, 0xc000786500})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.1/pkg/internal/controller/controller.go:223 +0x357
```

machine spec:
```
oc get IBMPowerVSMachine example-bfvkb  -n clusters-example -o yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: IBMPowerVSMachine
metadata:
  annotations:
    cluster.x-k8s.io/cloned-from-groupkind: IBMPowerVSMachineTemplate.infrastructure.cluster.x-k8s.io
    cluster.x-k8s.io/cloned-from-name: example
    hypershift.openshift.io/nodePoolPlatformMachineTemplate: '{"template":{"spec":{"serviceInstanceID":"a261b28e-c01b-48c4-9c50-c784c0e6143a","imageRef":{"name":"411.85.202203181612-0"},"sysType":"s922","procType":"shared","processors":"0.5","memory":"32","network":{"id":"e5398542-2881-46b7-9b9f-859acc1c5eaf"}}}}'
  creationTimestamp: "2022-04-07T07:06:17Z"
  generation: 1
  labels:
    cluster.x-k8s.io/cluster-name: example-9m5bw
    example-9m5bw-example-example: example-9m5bw-example-example
    machine-template-hash: "2984260431"
  managedFields:
  - apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:cluster.x-k8s.io/cloned-from-groupkind: {}
          f:cluster.x-k8s.io/cloned-from-name: {}
          f:hypershift.openshift.io/nodePoolPlatformMachineTemplate: {}
        f:labels:
          .: {}
          f:cluster.x-k8s.io/cluster-name: {}
          f:example-9m5bw-example-example: {}
          f:machine-template-hash: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"d2ca83d4-4e15-4dd6-b4f7-992e12aa6ede"}: {}
      f:spec:
        .: {}
        f:imageRef:
          .: {}
          f:name: {}
        f:memory: {}
        f:network:
          .: {}
          f:id: {}
        f:procType: {}
        f:processors: {}
        f:serviceInstanceID: {}
        f:sysType: {}
    manager: manager
    operation: Update
    time: "2022-04-07T07:06:18Z"
  name: example-bfvkb
  namespace: clusters-example
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Machine
    name: example-9m5bw-example-example-6fd86b4875-wqznx
    uid: d2ca83d4-4e15-4dd6-b4f7-992e12aa6ede
  resourceVersion: "510539"
  uid: 413a2495-9e2c-456e-bab3-b9d8fdcf12fc
spec:
  imageRef:
    name: 411.85.202203181612-0
  memory: "32"
  network:
    id: e5398542-2881-46b7-9b9f-859acc1c5eaf
  procType: shared
  processors: "0.5"
  serviceInstanceID: a261b28e-c01b-48c4-9c50-c784c0e6143a
  sysType: s922
```

machines are pending state
```
oc get IBMPowerVSMachine  -n clusters-example
NAME            CLUSTER         READY   STATE   HEALTH
example-bfvkb   example-9m5bw
example-bqqxg   example-9m5bw
example-tw5p5   example-9m5bw
---------------------------------------------------------------------------------------------
```
What did you expect to happen:
Capi-provider  pod shouldn't restarts.

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/cloud/scope/powervs_image.go
+++ b/cloud/scope/powervs_image.go
@@ -63,12 +63,14 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 	scope = &PowerVSImageScope{}
 
 	if params.Client == nil {
-		return nil, errors.New("failed to generate new scope from nil Client")
+		err = errors.New("failed to generate new scope from nil Client")
+		return
 	}
 	scope.client = params.Client
 
 	if params.IBMPowerVSImage == nil {
-		return nil, errors.New("failed to generate new scope from nil IBMPowerVSImage")
+		err = errors.New("failed to generate new scope from nil IBMPowerVSImage")
+		return
 	}
 	scope.IBMPowerVSImage = params.IBMPowerVSImage
 
@@ -79,24 +81,28 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 
 	helper, err := patch.NewHelper(params.IBMPowerVSImage, params.Client)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to init patch helper")
+		err = errors.Wrap(err, "failed to init patch helper")
+		return
 	}
+	scope.patchHelper = helper
 
 	spec := params.IBMPowerVSImage.Spec
 
 	auth, err := authenticator.GetAuthenticator()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get authenticator")
+		err = errors.Wrap(err, "failed to get authenticator")
+		return
 	}
 
 	account, err := servicesutils.GetAccount(auth)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get account")
+		err = errors.Wrap(err, "failed to get account")
+		return
 	}
 
 	rc, err := resourcecontroller.NewService(resourcecontroller.ServiceOptions{})
 	if err != nil {
-		return nil, err
+		return
 	}
 
 	res, _, err := rc.GetResourceInstance(
@@ -104,12 +110,14 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 			ID: core.StringPtr(spec.ServiceInstanceID),
 		})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get resource instance")
+		err = errors.Wrap(err, "failed to get resource instance")
+		return
 	}
 
 	region, err := utils.GetRegion(*res.RegionID)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get region")
+		err = errors.Wrap(err, "failed to get region")
+		return
 	}
 
 	options := powervs.ServiceOptions{
@@ -124,16 +132,12 @@ func NewPowerVSImageScope(params PowerVSImageScopeParams) (scope *PowerVSImageSc
 	c, err := powervs.NewService(options)
 
 	if err != nil {
-		return nil, fmt.Errorf("failed to create NewIBMPowerVSClient")
+		err = fmt.Errorf("failed to create NewIBMPowerVSClient")
+		return
 	}
+	scope.IBMPowerVSClient = c
 
-	return &PowerVSImageScope{
-		Logger:           params.Logger,
-		client:           params.Client,
-		IBMPowerVSClient: c,
-		IBMPowerVSImage:  params.IBMPowerVSImage,
-		patchHelper:      helper,
-	}, nil
+	return scope, nil
 }
 
 func (i *PowerVSImageScope) ensureImageUnique(imageName string) (*models.ImageReference, error) {

--- a/go.mod
+++ b/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.19.0
 	github.com/pkg/errors v0.9.1
-	github.com/ppc64le-cloud/powervs-utils v0.0.0-20210106101518-5d3f965b0344
+	github.com/ppc64le-cloud/powervs-utils v0.0.0-20210415051532-4cdd6a79c8fa
 	github.com/spf13/pflag v1.0.5
 	k8s.io/api v0.23.5
 	k8s.io/apimachinery v0.23.5

